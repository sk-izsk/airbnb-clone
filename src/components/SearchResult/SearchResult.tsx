import { Box, makeStyles, Typography } from '@material-ui/core';
import React from 'react';
import { AiFillStar } from 'react-icons/ai';
import { GrFavorite } from 'react-icons/gr';
import { CustomTheme } from '../../theme/muiTheme';

export interface SearchResultProps {
  img?: string;
  location?: string;
  title?: string;
  description?: string;
  star?: number;
  price?: string;
  total?: string;
}

const useStyles = makeStyles((theme: CustomTheme) => ({
  searchResult: {
    display: 'flex',
    position: 'relative',
    margin: theme.spacing(2.5),
    padding: theme.spacing(2.5),
    borderBottom: `1px solid lightgray`,
    cursor: 'pointer',
    '&:hover': {
      opacity: 0.8,
    },
  },
  img: {
    height: theme.spacing(25),
    width: theme.spacing(43.75),
    overflow: 'hidden',
    borderRadius: 10,
  },
  searchResultHeart: {
    position: 'absolute',
    top: theme.spacing(2.5),
    right: theme.spacing(5),
  },
  searchResultInfo: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    paddingLeft: theme.spacing(2.5),
  },
  searchResultInfoTop: {
    width: '40vw',
  },
  searchResultInfoBottom: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  searchResultStars: {
    display: 'flex',
    alignItems: 'center',
  },
  searchResultStar: {},
  searchResultsPrice: {
    position: 'absolute',
    bottom: theme.spacing(2.5),
    right: theme.spacing(3.75),
  },
  title: {
    fontWeight: 300,
    marginTop: theme.spacing(1.25),
  },
  normalText: {
    marginTop: theme.spacing(1.25),
    fontSize: 13,
    color: 'gray',
  },
  total: {
    textAlign: 'right',
  },
}));

const SearchResult: React.FC<SearchResultProps> = ({ description, img, location, price, star, title, total }) => {
  const classes = useStyles();
  return (
    <Box className={classes.searchResult}>
      <img className={classes.img} alt='' src={img} />
      <GrFavorite className={classes.searchResultHeart} />
      <Box className={classes.searchResultInfo}>
        <Box className={classes.searchResultInfoTop}>
          <Typography className={classes.normalText}>{location}</Typography>
          <Typography className={classes.title} variant='h4'>
            {title}
          </Typography>
          <Typography className={classes.normalText}>______</Typography>
          <Typography className={classes.normalText}>{description}</Typography>
        </Box>
        <Box className={classes.searchResultInfoBottom}>
          <Box className={classes.searchResultStars}>
            <AiFillStar className={classes.searchResultStar} color='#ff7779' />
            <Typography>{star}</Typography>
          </Box>
          <Box className={classes.searchResultsPrice}>
            <Typography variant='h3'>{price}</Typography>
            <Typography className={classes.total}>{total}</Typography>
          </Box>
        </Box>
      </Box>
    </Box>
  );
};

export { SearchResult };
