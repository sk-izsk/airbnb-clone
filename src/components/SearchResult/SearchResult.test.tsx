import { render } from '@testing-library/react';
import React from 'react';
import { SearchResult } from './SearchResult';
const mockData = {
  img: 'mock src',
  description: 'mock desc',
  price: 'mock price',
  title: 'mock title',
  star: 5,
  location: 'mock location',
  total: 'mock total',
};
test('SearchResult renders correctly', () => {
  const { asFragment } = render(<SearchResult {...mockData} />);
  expect(asFragment()).toMatchSnapshot();
});
