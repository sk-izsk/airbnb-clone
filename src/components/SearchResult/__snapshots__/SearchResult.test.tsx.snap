// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-13 makeStyles-searchResult-1"
  >
    <img
      alt=""
      class="makeStyles-img-2"
      src="mock src"
    />
    <svg
      class="makeStyles-searchResultHeart-3"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1,8.4 C1,4 4.5,3 6.5,3 C9,3 11,5 12,6.5 C13,5 15,3 17.5,3 C19.5,3 23,4 23,8.4 C23,15 12,21 12,21 C12,21 1,15 1,8.4 Z"
        fill="none"
        stroke="#000"
        stroke-width="2"
      />
    </svg>
    <div
      class="MuiBox-root MuiBox-root-14 makeStyles-searchResultInfo-4"
    >
      <div
        class="MuiBox-root MuiBox-root-15 makeStyles-searchResultInfoTop-5"
      >
        <p
          class="MuiTypography-root makeStyles-normalText-11 MuiTypography-body1"
        >
          mock location
        </p>
        <h4
          class="MuiTypography-root makeStyles-title-10 MuiTypography-h4"
        >
          mock title
        </h4>
        <p
          class="MuiTypography-root makeStyles-normalText-11 MuiTypography-body1"
        >
          ______
        </p>
        <p
          class="MuiTypography-root makeStyles-normalText-11 MuiTypography-body1"
        >
          mock desc
        </p>
      </div>
      <div
        class="MuiBox-root MuiBox-root-16 makeStyles-searchResultInfoBottom-6"
      >
        <div
          class="MuiBox-root MuiBox-root-17 makeStyles-searchResultStars-7"
        >
          <svg
            class="makeStyles-searchResultStar-8"
            color="#ff7779"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 119, 121);"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            5
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-18 makeStyles-searchResultsPrice-9"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            mock price
          </h3>
          <p
            class="MuiTypography-root makeStyles-total-12 MuiTypography-body1"
          >
            mock total
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
